massa-ledger-worker/src/ledger_db.rs

```diff
@@ -572,36 +694,37 @@ mod tests {
     /// Functional test of LedgerDB
     #[test]
     fn test_ledger_db() {
-        // init addresses
-        let pub_a = KeyPair::generate().get_public_key();
-        let pub_b = KeyPair::generate().get_public_key();
-        let a = Address::from_public_key(&pub_a);
-        let b = Address::from_public_key(&pub_b);
-        let (db, data) = init_test_ledger(a);
+        let addr = Address::from_public_key(&KeyPair::generate().get_public_key());
+        let (db, data) = init_test_ledger(addr);
         let amount_deserializer =
             AmountDeserializer::new(Included(Amount::MIN), Included(Amount::MAX));
-        // first assert
-        assert!(db.get_sub_entry(&a, LedgerSubEntry::ParBalance).is_some());
+
+        // check initial state and entry update
+        assert!(db
+            .get_sub_entry(&addr, LedgerSubEntry::ParBalance)
+            .is_some());
         assert_eq!(
             amount_deserializer
                 .deserialize::<DeserializeError>(
-                    &db.get_sub_entry(&a, LedgerSubEntry::ParBalance).unwrap()
+                    &db.get_sub_entry(&addr, LedgerSubEntry::ParBalance).unwrap()
                 )
                 .unwrap()
                 .1,
-            Amount::from_mantissa_scale(21, 0)
+            Amount::from_str("21").unwrap()
         );
-        assert!(db.get_sub_entry(&b, LedgerSubEntry::ParBalance).is_none());
-        assert_eq!(data, db.get_entire_datastore(&a));
+        assert_eq!(data, db.get_entire_datastore(&addr));
 
         // delete entry
-        let mut batch = WriteBatch::default();
-        db.delete_entry(&a, &mut batch);
+        let mut batch = LedgerBatch::new(db.get_ledger_hash());
```

Comment:
Should we at least check that the hash != Hash::from_bytes(&[0; 32]) ? Or maybe add a const with an expected value?